### PR TITLE
added login to api logic

### PR DESCRIPTION
**Pull request by BlesyMathew**
Signed-off-by: BLESSY MATHEW <74411288+BlesyMathew@users.noreply.github.com>

### CODE REVIEW

File: src/components/layouts/authLayout.tsx [hunk 1]
```diff
@@ -2,28 +2,32 @@ import { Avatar, AvatarFallback, AvatarImage } from "../ui/avatar";
...
 
 const AuthLayout = (): JSX.Element => {
   const navigate = useNavigate();
   return (
     <div
       style={{
-        backgroundImage:
-          "url('https://mdbootstrap.com/img/Photos/Others/images/76.jpg",
-        height: "130vh",
+        backgroundColor: "black",
```
Comment (shyluthomas) [inline src/components/layouts/authLayout.tsx]: inline style

File: src/components/Modal/CommonModal.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import React, { useState } from "react";
+
+import { ModalProps } from "@/types/user";
+
+type CommonModalProps = { message: ModalProps };
+
+export default function CommonModal({ message }: CommonModalProps) {
+  const [dialog, setDialog] = useState(message.status);
+  const status = message.status;
+  console.log("modalll", status);
```
Comment (shyluthomas) [inline src/components/Modal/CommonModal.tsx]: Remove console

File: src/components/layouts/ModalHOC.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import CommonModal from "../Modal/CommonModal";
+import { useAppSelector } from "@/hooks/hooks";
+
+export default function ModalHOC() {
+  const modalMessage = useAppSelector((state) => state.userReducer.dialog);
+  console.log("modalMessage", modalMessage);
```
Comment (shyluthomas) [inline src/components/layouts/ModalHOC.tsx]: console

File: src/components/layouts/authLayout.tsx [hunk 4]
```diff
@@ -2,28 +2,27 @@ import { Avatar, AvatarFallback, AvatarImage } from "../ui/avatar";
 import { Menubar, MenubarMenu, MenubarTrigger } from "../ui/menubar";
 
 import { Outlet } from "react-router-dom";
+// import { Toaster } from "../ui/sonner";
```
Comment (shyluthomas) [inline src/components/layouts/authLayout.tsx]: comments

File: src/components/layouts/fullLayout.tsx [hunk 5]
```diff
@@ -1,10 +1,63 @@
...
+              </DropdownMenuTrigger>
+              <DropdownMenuContent className="bg-white">
+                <DropdownMenuLabel>My Account</DropdownMenuLabel>
+                <DropdownMenuSeparator />
+                <DropdownMenuItem>Profile</DropdownMenuItem>
+                <DropdownMenuItem>Billing</DropdownMenuItem>
+                <DropdownMenuItem>Team</DropdownMenuItem>
+                <DropdownMenuItem
+                  onClick={() => {
+                    helpers.removeAuthToken();
```
Comment (shyluthomas) [inline src/components/layouts/fullLayout.tsx]: do not do logic in components, do saga fn and dispatch here

File: src/pages/user/newUser.tsx [hunk 6]
```diff
@@ -43,43 +45,27 @@ const createdUser = useAppSelector((state)=> state.userReducer.userCreation);
...
   return (
     <div className="flex mt-16 justify-center items-center">
-      <Card className="w-[450px]">
+      <Card className="w-[550px]">
         <form onSubmit={handleSubmit(submitLogin)}>
           <CardHeader>
             <CardTitle>Sign Up</CardTitle>
             <CardDescription>New User registeration</CardDescription>
-            <div>{createdUser.status}</div>
+            <ModalHOC />
```
Comment (shyluthomas) [inline src/pages/user/newUser.tsx]: Add this HOC in Layout component: ie, you can use in whole app

File: src/pages/user/newUser.tsx [hunk 7]
```diff
@@ -233,7 +219,7 @@ const createdUser = useAppSelector((state)=> state.userReducer.userCreation);
                 )}
               </div>
               <div className="flex flex-col space-y-1.5">
-                <Label htmlFor="roleId">Role ID</Label>
+                {/* <Label htmlFor="roleId">Role ID</Label> */}
```
Comment (shyluthomas) [inline src/pages/user/newUser.tsx]: remove comments

File: src/pages/user/userLogin.tsx [hunk 8]
```diff
@@ -1,68 +1,106 @@
...
+import { useNavigate } from "react-router-dom";
 
 const UserLogin = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const loggedUser = useAppSelector((state) => state.userReducer.auth);
+  const loggedIn = loggedUser.isAuthenticated;
+  const username = useAppSelector((state) => {
+    state.userReducer.userLogin;
+  });
+  console.log("username", username);
```
Comment (shyluthomas) [inline src/pages/user/userLogin.tsx]: console and comments

File: src/store/saga/mainSaga.ts [hunk 9]
```diff
@@ -19,25 +19,43 @@ function* _getAuthCheck(action: any) {
...
   }
 }
 
+function* _setLogin(action: UserAction): any {
+  try {
+    const userData = action.payload;
+    const response = yield userService.loginUser(userData);
+    //setting auth from the api response
+    if (response.status === 200) {
+      helpers.setAuthToken(response.data.token);
```
Comment (shyluthomas) [inline src/store/saga/mainSaga.ts]: you need to add yield for this fn call

File: src/store/saga/mainSaga.ts [hunk 10]
```diff
@@ -19,25 +19,43 @@ function* _getAuthCheck(action: any) {
...
+      yield put(user.setAuth({ isAuthenticated: true, user: response.data }));
+    } else {
+      yield put(user.setAuth({ isAuthenticated: false }));
+    }
+  } catch (e) {
+    yield put(user.setAuth({ isAuthenticated: false }));
+    console.log("first");
```
Comment (shyluthomas) [inline src/store/saga/mainSaga.ts]: what the fck

File: src/utils/helpers.ts [hunk 11]
```diff
@@ -16,6 +16,9 @@ const helpers = {
       return false;
     }
   },
+  removeAuthToken: () => {
+    sessionStorage.removeItem("accessToken");
```
Comment (shyluthomas) [inline src/utils/helpers.ts]: it should async fn

**Review by shyluthomas — CHANGES_REQUESTED** [review]

File: src/components/layouts/authLayout.tsx (hunk 1 above)
Comment (BlesyMathew) [reply src/components/layouts/authLayout.tsx]: this is the outdated code mahn

**Review by shyluthomas — APPROVED** [review]

**Pull request merged**